ledger/src/blockstore.rs

```diff
@@ -1134,7 +1123,10 @@ impl Blockstore {
 
         // Commit step: commit all changes to the mutable structures at once, or none at all.
         // We don't want only a subset of these changes going through.
-        write_batch.put_bytes::<cf::ShredData>((slot, index), &shred.payload)?;
+        let path = self.data_shred_path(slot, index);
+        let shred_path = Path::new(&path);
+        let mut f = fs::File::create(shred_path)?;
+        f.write_all(&shred.payload)?;
```

Comment:
Now that these data/coding insert operations are no longer atomic with respect to the `SlotMeta`'s and other metadata, there may be  some additional scenarios we need to reason through.

For instance: what happens if we insert the data shred, but crash before the SlotMeta is committed?

Then repair's data iterator won't find this shred is missing, so the validator will never request this shred again. So unless the network happens to give us this shred again, the SlotMeta may never update.

Reply:
With enough erasure coding it’s not an issue :)

Reply:
We don't send erasure through repair. Moreover, the way we know erasure is ready to be performed is through the `Index` metadata tracking the number or erasure/data shreds. This `Index` is updated in the same batch as the `SlotMeta`, so it also won't update if we crash here.

Reply:
I mean, the block will recover even with errors, right?

Reply:
I get the first `n` of the data shreds of a block, then crash before I can commit the updates to the `SlotMeta`. 

When I reboot I check `SlotMeta` and see there are no playable shreds so ReplayStage stalls.

Meanwhile repair does not ask for the first `n` shreds because they exist in the ledger.

How would this recover?

Reply:
Where do we check for playable shreds?

Reply:
`get_slot_entries() -> get_slot_entries_with_shred_info()`

Reply:
@carllin ill reconcile the data in that funciton